packages/api/models/collections.js

```diff
@@ -88,7 +104,15 @@ class Collection extends Manager {
       );
     }
 
-    await super.delete({ name, version });
+    // Since the `create` method uses the collection's `dataType` when calling
+    // `CollectionConfigStore.put`, we must also use `dataType` to delete it
+    // from the store.  However, we have only the collection's name and version,
+    // so we need to retrieve the full collection object in order to retrieve
+    // its dataType.
+    const { dataType } = await super.get({ name, version });
```

Comment:
It's not a huge deal, but we probably want to use `this.get` instead of `super.get`. Right now, there's no difference, but if for some reason we did override `get` for this class in the future, we'd want to use that instead of the method on the superclass

Reply:
Good catch. I'll change that. I think I was in a `super` mindset, given that `super.delete` is called below.

Reply:
Yeah, we want to use `super.delete` because we're already inside the overridden `delete` method and want to call the superclass method for the basic delete functionality, but we want `this.get` to capture any possible future superclass override